things/mocks/auth.go

```diff
@@ -42,15 +48,18 @@ func (svc authServiceMock) Issue(ctx context.Context, in *mainflux.IssueReq, opt
 }
 
 func (svc authServiceMock) Authorize(ctx context.Context, req *mainflux.AuthorizeReq, _ ...grpc.CallOption) (r *mainflux.AuthorizeRes, err error) {
-	if email, ok := svc.users["token"]; ok {
-		if email == req.GetSub() {
+	for _, policy := range svc.policies[req.GetSub()] {
+		if policy.Relation == req.GetAct() && policy.Object == req.GetObj() {
 			return &mainflux.AuthorizeRes{Authorized: true}, nil
 		}
 	}
 	return nil, things.ErrAuthorization
 }
 
 func (svc authServiceMock) AddPolicy(ctx context.Context, in *mainflux.AddPolicyReq, opts ...grpc.CallOption) (*mainflux.AddPolicyRes, error) {
+	obj := in.GetObj()
+	obj = obj[len(obj)-3:]
```

Comment:
Magic numbers. What are these constant numbers? Should they be declared as a constant somewhere. Also - please add a comment of what's going on here.

Reply:
While creating a `Thing`, `CreateThings` on the service layer assigns a new ID for the thing [as here.](https://github.com/mainflux/mainflux/blob/f4312aef8ac288f97c4db0e2dcb4e70475e620bf/things/service.go#L170)

Then, the `mock Things repository` saves this thing ID by padding [as here](https://github.com/mainflux/mainflux/blob/f4312aef8ac288f97c4db0e2dcb4e70475e620bf/things/mocks/things.go#L58)

Since we are adding policies within the Service layer, we are storing them as a full ID which is eventually not compatible with the one inside of the `mock things repository`.

Therefore, I used the padded part while adding a policy. That's why I used 
```go
obj = obj[len(obj)-3:]
```

Reply:
OK, but then needs a comment.